gateway/gateway_test.go

```diff
@@ -504,317 +116,614 @@ func TestPretty404(t *testing.T) {
 	} {
 		testName := fmt.Sprintf("%s %s", test.path, test.accept)
 		t.Run(testName, func(t *testing.T) {
-			var c http.Client
-			req, err := http.NewRequest("GET", ts.URL+test.path, nil)
-			assert.NoError(t, err)
+			req := mustNewRequest(t, "GET", ts.URL+test.path, nil)
 			req.Header.Add("Accept", test.accept)
 			req.Host = host
-			resp, err := c.Do(req)
-			assert.NoError(t, err)
+			resp := mustDo(t, req)
 			defer resp.Body.Close()
-			assert.Equal(t, test.status, resp.StatusCode)
+			require.Equal(t, test.status, resp.StatusCode)
 			body, err := io.ReadAll(resp.Body)
-			assert.NoError(t, err)
+			require.NoError(t, err)
 			if test.text != "" {
-				assert.Equal(t, test.text, string(body))
+				require.Equal(t, test.text, string(body))
 			}
 		})
 	}
 }
 
-func TestBrowserErrorHTML(t *testing.T) {
+func TestHeaders(t *testing.T) {
+	t.Parallel()
+
 	ts, _, root := newTestServerAndNode(t, nil)
-	t.Logf("test server url: %s", ts.URL)
 
-	t.Run("plain error if request does not have Accept: text/html", func(t *testing.T) {
-		t.Parallel()
+	var (
+		dirCID   = "bafybeihta5xfgxcmyxyq6druvidc7es6ogffdd6zel22l3y4wddju5xxsu"
+		dirPath  = "/ipfs/" + root.String() + "/subdir/"
+		dirRoots = "bafybeifhvgr4ufgwpoj2iyrymrigjhpexvsyfm2elafebkmrei4skunihe," + dirCID
```

Comment:
:thought_balloon: The setup with single `./testdata/fixtures.car` starts to feel a bit brittle.

Every time `./testdata/fixtures.car` is updated when we add new tests  → the  `root.String()` CID changes here  → the first CID  in `dirRoots` below will no longer match, and someone will have to update these to new correct values, which introduces risk of test regression if not careful.

Would be less brittle if we split `fixtures.car` and have a separate `./testdata/roots-header.car` fixture for this + hardcoded explicit CIDs here, so we never have to change this, so if tests fail in the future, we know it's a bug, and not  maintenance related to updating `root.String()`?